andela/suyabay
# #115755943  Create all the boilerplate Restful API endpoints for Suyabay.

#### What does this PR do?

Create all the boilerplate Restful API endpoints for Suyabay. 
#### Description of Task to be completed?
- Create user endpoints and the callback functions.
  - Get all users GET /api/v1/users
  - Get a single  user GET /api/v1/users/{id}
  - Post a user POST /api/v1/users
  - Update user PUT /api/v1/users/{id}
- Create channel endpoints and the callback functions
  - Get all channels GET /api/v1/channels
  - Get a single  channel GET /api/v1/channels/{id}
  - Post a channel POST /api/v1/channels
  - Update a channel PUT /api/v1/channels/{id}
- Create channel episodes endpoints and the callback functions.
  - Get all channel episodes GET /api/v1/channels/{channel_id}/episodes
  - Get a single channel episode  GET /api/v1/channels/{channel_id}/episodes/{episode_id}
  - Post a channel episode POST /api/v1/channels/{channel_id}/episodes
  - Update a channel PUT /api/v1/channels/{channel_id}/episodes/{episode_id}
- Create episode comments endpoints and the callback functions.
  - Get all episode comments GET /api/v1/episodes/{episode_id}/comments
  - Get a single  episode comment  GET /api/v1/episodes/{episode_id}/comments/{comment_id}
  - Post an episode comments  POST /api/v1/episodes/{episode_id}/comments
- Create user favourites endpoints and the callback functions
  - Get all user favourites GET /api/v1/users/{username}/favourites
  - Get a single favourite  GET /api/v1/users/{username}/favourites/{favourite_id}
  - Post a favourite  POST /api/v1/users/{username}/favourites
#### How should this be manually tested?
- You need to add Postman as an add-on extension to your browser
- Input the endpoint you want to test in the url section  e.g. /api/v1/users
- In the response body, you will be presented with a Json  e.g. {"message" : "return all users" } and the response header should contain the status code 200 OK.
#### What are the relevant pivotal tracker stories?
#115755943
